h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstPivot.java

```diff
@@ -0,0 +1,79 @@
+package water.rapids.ast.prims.mungers;
+
+import water.*;
+import water.fvec.*;
+import water.rapids.Rapids;
+import water.rapids.Env;
+import water.rapids.vals.ValFrame;
+import water.rapids.ast.AstPrimitive;
+import water.rapids.ast.AstRoot;
+import water.util.VecUtils;
+
+public class AstPivot extends AstPrimitive {
+  @Override
+  public String[] args() {
+    return new String[]{"ary", "index", "column", "value"}; //the array and name of columns
+  }
+
+  @Override
+  public int nargs() {
+    return 1 + 4;
+  } // (pivot ary index column value)
+
+  @Override
+  public String str() {
+    return "pivot";
+  }
+
+  @Override
+  public ValFrame apply(Env env, Env.StackHelp stk, AstRoot asts[]) {
+    Frame origfr = stk.track(asts[1].exec(env)).getFrame();
+    String _index = stk.track(asts[2].exec(env)).getStr();
+    String _column = stk.track(asts[3].exec(env)).getStr();
+    String _value = stk.track(asts[4].exec(env)).getStr();
+    Key k1 = Key.<Frame>make();
+    Frame fr = origfr.subframe(new String[]{_index,_column,_value}).deepCopy(k1.toString());
```

Comment:
so my problem was this, if i put the original frame into the DKV , which I need to do to get Rapids to run,  when I delete the frame at the end (so I dont get leaked keys)  the original frame is deleted, which I dont want to do.  
In python if I say:
  dfpivoted = df.pivot("index","column","value")

If I use the original frame and delete it at the end, df is no longer there.

Is there a better way to remove the key from the DKV?